File: src-v8/Mike/Pic Admin/Form1.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.WinForms

Private Class CategoryNode
  Inherits TreeNode
  
  Public CategoryID As Integer
  
  Public Sub CategoryNode(ByVal CategoryName As String, ByVal CategoryID As String)
    
  End Sub
  
  
End Class


Public Class Form1
  Inherits System.WinForms.Form
  
  Public Sub New()
    MyBase.New()
    
    Form1 = Me
    
    'This call is required by the Win Form Designer.
    InitializeComponent()
    
    'TODO: Add any initialization after the InitializeComponent() call
    
    cmdCategory.FillDataSet(dsCategory)
    
    Dim rootNode As TreeNode = tvCat.Nodes.Add("Category")
    
    LoadCategories(dsCategory, 1, rootNode)
    
  End Sub
  
  'Form overrides dispose to clean up the component list.
  Public Overrides Sub Dispose()
    MyBase.Dispose()
    components.Dispose()
  End Sub
  
  Public Sub LoadCategories(ByVal dsData As CdsCategory, ByVal intParentID As Integer, ByVal ParentNode As treenode)
    
    Dim li As ListItem, dv As DataView, child As TreeNode, i As Integer
    
    dv = New DataView(dsData.Tables("Category"))
    dv.RowFilter = "ParentID = " & intParentID & " AND CategoryID <> " & intParentID
    
    For i = 0 To dv.Count - 1
      child = ParentNode.Nodes.Add(dv(i)("CategoryName").ToString)
      LoadCategories(dsData, CInt(dv(i)("CategoryID")), child)
    Next i
    
    li = Me.lvSubCat.ListItems.Add("Add new category")
    
    
  End Sub
  
  
#Region " Windows Form Designer generated code "
  
  'Required by the Windows Form Designer
  Private components As System.ComponentModel.Container
  Private WithEvents dsCategory As Pic_Admin.CdsCategory
  
  Private WithEvents cmdCategory As System.Data.SQL.SQLDataSetCommand
  Private chName As System.WinForms.ColumnHeader
  
  Private WithEvents cnPicDB As System.Data.SQL.SQLConnection
  
  
  
  
  
  
  Private WithEvents lvSubCat As System.WinForms.ListView
  Private WithEvents tvCat As System.WinForms.TreeView
  
  Dim WithEvents Form1 As System.WinForms.Form
  
  'NOTE: The following procedure is required by the Windows Form Designer
  'It can be modified using the Windows Form Designer.  
  'Do not modify it using the code editor.
  Private Sub InitializeComponent()
    Me.components = New System.ComponentModel.Container()
    Me.chName = New System.WinForms.ColumnHeader()
    Me.lvSubCat = New System.WinForms.ListView()
    Me.dsCategory = New Pic_Admin.CdsCategory()
    Me.tvCat = New System.WinForms.TreeView()
    Me.cmdCategory = New System.Data.SQL.SQLDataSetCommand()
    Me.cnPicDB = New System.Data.SQL.SQLConnection()
    
    '@design Me.TrayHeight = 90
    '@design Me.TrayLargeIcon = False
    '@design Me.TrayAutoArrange = True
    chName.Text = "Name"
    chName.Width = 206
    chName.TextAlign = System.WinForms.HorizontalAlignment.Left
    
    lvSubCat.Location = New System.Drawing.Point(216, 32)
    lvSubCat.Size = New System.Drawing.Size(312, 344)
    lvSubCat.FullRowSelect = True
    lvSubCat.View = System.WinForms.View.Report
    lvSubCat.ForeColor = System.Drawing.SystemColors.WindowText
    lvSubCat.TabIndex = 1
    lvSubCat.Anchor = System.WinForms.AnchorStyles.All
    Dim a__1(1) As System.WinForms.ColumnHeader
    a__1(0) = chName
    lvSubCat.Columns.All = a__1
    
    '@design dsCategory.SetLocation(New System.Drawing.Point(194, 7))
    dsCategory.Prefix = ""
    dsCategory.DataSetName = "CdsCategory"
    dsCategory.Locale = New System.Globalization.CultureInfo("en-US")
    dsCategory.Namespace = "http://www.tempuri.org/CdsCategory.xsd"
    dsCategory.Category.Prefix = ""
    dsCategory.Category.TableName = "Category"
    
    tvCat.Location = New System.Drawing.Point(8, 32)
    tvCat.Size = New System.Drawing.Size(200, 344)
    tvCat.TabIndex = 0
    tvCat.Anchor = System.WinForms.AnchorStyles.TopBottomLeft
    tvCat.ShowRootLines = False
    
    '@design cmdCategory.SetLocation(New System.Drawing.Point(89, 7))
    Dim a__2() As System.Data.SQL.SQLParameter
    cmdCategory.UpdateCommand = New System.Data.SQL.SQLCommand(cnPicDB, "UPDATE Category SET ParentID = ?,  CategoryName = ?,  FullCategory = ?,  AccessLevel = ? WHERE (CategoryID = ?) ;SELECT CategoryID,  ParentID,  CategoryName,  FullCategory,  AccessLevel FROM Category WHERE (CategoryID = ?) ", System.Data.CommandType.Text, a__2, System.Data.UpdateRowSource.Both)
    Dim a__3() As System.Data.SQL.SQLParameter
    cmdCategory.DeleteCommand = New System.Data.SQL.SQLCommand(cnPicDB, "DELETE FROM Category WHERE (CategoryID = ?) ", System.Data.CommandType.Text, a__3, System.Data.UpdateRowSource.Both)
    Dim a__4() As System.Data.SQL.SQLParameter
    cmdCategory.InsertCommand = New System.Data.SQL.SQLCommand(cnPicDB, "INSERT INTO Category( ParentID,  CategoryName,  FullCategory,  AccessLevel ) VALUES( ?,  ?,  ?,  ? ) ;SELECT CategoryID,  ParentID,  CategoryName,  FullCategory,  AccessLevel FROM Category WHERE (CategoryID = @@IDENTITY) ", System.Data.CommandType.Text, a__4, System.Data.UpdateRowSource.Both)
    Dim a__5() As System.Data.SQL.SQLParameter
    cmdCategory.SelectCommand = New System.Data.SQL.SQLCommand(cnPicDB, "SELECT CategoryID,  ParentID,  CategoryName,  FullCategory,  AccessLevel FROM Category ", System.Data.CommandType.Text, a__5, System.Data.UpdateRowSource.Both)
    Dim a__6(5) As System.Data.Internal.DataColumnMapping
    a__6(0) = New System.Data.Internal.DataColumnMapping("CategoryID", "CategoryID")
    a__6(1) = New System.Data.Internal.DataColumnMapping("ParentID", "ParentID")
    a__6(2) = New System.Data.Internal.DataColumnMapping("CategoryName", "CategoryName")
    a__6(3) = New System.Data.Internal.DataColumnMapping("FullCategory", "FullCategory")
    a__6(4) = New System.Data.Internal.DataColumnMapping("AccessLevel", "AccessLevel")
    Dim a__7(1) As System.Data.Internal.DataTableMapping
    a__7(0) = New System.Data.Internal.DataTableMapping("Table", "Category", a__6)
    cmdCategory.TableMappings.All = a__7
    
    '@design cnPicDB.SetLocation(New System.Drawing.Point(7, 7))
    cnPicDB.ConnectionString = "data source=cheese; initial catalog=picdb; integrated security=sspi;"
    
    Me.Text = "Form1"
    Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
    Me.ClientSize = New System.Drawing.Size(536, 381)
    
    Me.Controls.Add(lvSubCat)
    Me.Controls.Add(tvCat)
  End Sub
  
#End Region
  
  Public Sub lvSubCat_DoubleClick(ByVal sender As Object, ByVal e As System.EventArgs) Handles lvSubCat.DoubleClick
    
    Dim li As ListItem, intResult As Integer
    li = lvSubCat.SelectedItems(0)
    
    Dim f As New Category()
    Try
      If f.ShowDialog(Me) = DialogResult.OK Then
        Me.tvCat.Nodes.Add(f.CategoryName)
        
        
        
        msgbox("Dialog Result is OK", Microsoft.VisualBasic.MsgBoxStyle.Information)
      End If
      
      
    Catch ec As Exception
      msgbox(ec.ToString, Microsoft.VisualBasic.MsgBoxStyle.Information)
    End Try
    
    
  End Sub
  
  
End Class

File: Pictures/Admin/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using msn2.net.Pictures.Controls;

namespace msn2.net.Pictures
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class Form1 : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Button button1;
		private msn2.net.Pictures.Controls.EditPictureLink editPictureLink1;
		private msn2.net.Pictures.Controls.EditCategoryLink editCategoryLink1;
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Label label2;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public Form1()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.button1 = new System.Windows.Forms.Button();
			this.editPictureLink1 = new msn2.net.Pictures.Controls.EditPictureLink();
			this.editCategoryLink1 = new msn2.net.Pictures.Controls.EditCategoryLink();
			this.label1 = new System.Windows.Forms.Label();
			this.label2 = new System.Windows.Forms.Label();
			this.SuspendLayout();
			// 
			// button1
			// 
			this.button1.Location = new System.Drawing.Point(72, 56);
			this.button1.Name = "button1";
			this.button1.TabIndex = 0;
			this.button1.Text = "Main Form";
			this.button1.Click += new System.EventHandler(this.button1_Click);
			// 
			// editPictureLink1
			// 
			this.editPictureLink1.Location = new System.Drawing.Point(40, 160);
			this.editPictureLink1.Name = "editPictureLink1";
			this.editPictureLink1.PictureId = 2120;
			this.editPictureLink1.Size = new System.Drawing.Size(48, 16);
			this.editPictureLink1.TabIndex = 1;
			this.editPictureLink1.Load += new System.EventHandler(this.editPictureLink1_Load);
			// 
			// editCategoryLink1
			// 
			this.editCategoryLink1.CategoryId = 165;
			this.editCategoryLink1.Location = new System.Drawing.Point(176, 160);
			this.editCategoryLink1.Name = "editCategoryLink1";
			this.editCategoryLink1.SignalRefresh = false;
			this.editCategoryLink1.Size = new System.Drawing.Size(48, 16);
			this.editCategoryLink1.TabIndex = 2;
			this.editCategoryLink1.Load += new System.EventHandler(this.editCategoryLink1_Load);
			// 
			// label1
			// 
			this.label1.Location = new System.Drawing.Point(176, 128);
			this.label1.Name = "label1";
			this.label1.TabIndex = 3;
			this.label1.Text = "Category";
			// 
			// label2
			// 
			this.label2.Location = new System.Drawing.Point(24, 128);
			this.label2.Name = "label2";
			this.label2.TabIndex = 4;
			this.label2.Text = "Picture";
			// 
			// Form1
			// 
			this.ClientSize = new System.Drawing.Size(292, 266);
			this.Controls.Add(this.label2);
			this.Controls.Add(this.label1);
			this.Controls.Add(this.editCategoryLink1);
			this.Controls.Add(this.editPictureLink1);
			this.Controls.Add(this.button1);
			this.Name = "Form1";
			this.Text = "Form1";
			this.ResumeLayout(false);

		}
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
		//	Application.Run(new Form1());

            try
            {
                msn2.net.Pictures.Controls.fMain f = new msn2.net.Pictures.Controls.fMain();
                Application.Run(f);
            }
            catch (Exception ex)
            {
                ExceptionDialog dialog = new ExceptionDialog(ex);
                dialog.ShowDialog();
            }
        }

        private void button1_Click(object sender, System.EventArgs e)
		{
			msn2.net.Pictures.Controls.fMain f = new msn2.net.Pictures.Controls.fMain();
			f.Show();
		}

		private void editPictureLink1_Load(object sender, System.EventArgs e)
		{
		
		}

		private void editCategoryLink1_Load(object sender, System.EventArgs e)
		{
		
		}
	}
}
